src/main/java/com/ververica/CiBot.java

```diff
@@ -142,7 +142,8 @@ public CiBot(Core core, GitActionsImpl gitActions, int pollingIntervalInSeconds,
 	public void run() {
 		try {
 			Date lastUpdateDate = Date.from(Instant.now().minus(Duration.ofHours(backlogHours)));
-			while (true) {
+			boolean runContinuously = pollingIntervalInSeconds > 0;
+			do {
```

Comment:
let's move the inner `try` block into a separate method, and structure `run` like this:

```
if (runContinuously) {
	while (true) {
		internalRun()
		<sleep>
	}
} else {
	internalRun()
}
```